koku/subs/subs_data_messenger.py

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright 2023 Red Hat Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+import csv
+import json
+import logging
+import os
+import uuid
+from tempfile import mkdtemp
+
+from django.conf import settings
+
+from api.common import log_json
+from api.iam.models import Customer
+from kafka_utils.utils import delivery_callback
+from kafka_utils.utils import get_producer
+from masu.config import Config as masu_config
+from masu.prometheus_stats import KAFKA_CONNECTION_ERRORS_COUNTER
+from subs.subs_data_extractor import get_subs_s3_client
+
+LOG = logging.getLogger(__name__)
+
+
+class SUBSDataMessenger:
+    def __init__(self, context, schema_name, tracing_id):
+        self.context = context
+        self.tracing_id = tracing_id
+        self.schema_name = schema_name
+        self.s3_client = get_subs_s3_client()
+        subs_cust = Customer.objects.filter(schema_name=schema_name).first()
+        self.account_id = subs_cust.account_id
+        self.org_id = subs_cust.org_id
+        self.download_path = mkdtemp(prefix="subs")
+
+    def process_and_send_subs_message(self, upload_keys):
+        """
+        Takes a list of object keys, reads the objects from the S3 bucket and processes a message to kafka.
+        """
+        for i, obj_key in enumerate(upload_keys):
+            csv_path = f"{self.download_path}/subs_{self.tracing_id}_{i}.csv"
+            self.s3_client.download_file(settings.S3_SUBS_BUCKET_NAME, obj_key, csv_path)
+            with open(csv_path) as csv_file:
+                reader = csv.DictReader(csv_file)
+                LOG.info(
+                    log_json(
+                        self.tracing_id,
+                        msg="iterating over records and sending kafka messages",
+                        context=self.context,
+                    )
+                )
+                msg_count = 0
+                for row in reader:
+                    msg = self.build_subs_msg(
```

Comment:
I suppose this is fine for now, but are these the columns we will be getting for Azure and GCP too?

Reply:
We will need different columns when it comes time for Azure/GCP, I think for now this function can stay as is and when its time to add another provider, we can take into account what field names to use.